packages/core/test/unit/query-generator/add-index-query.test.ts

```diff
@@ -0,0 +1,282 @@
+import { Op } from '@sequelize/core';
+import { buildInvalidOptionReceivedError } from '@sequelize/core/_non-semver-use-at-your-own-risk_/utils/check.js';
+import { createSequelizeInstance, expectsql, sequelize } from '../../support';
+
+const dialect = sequelize.dialect;
+const notSupportedError = new Error(`addIndexesQuery is not supported in ${dialect.name} dialect.`);
+
+describe('QueryGenerator#addIndexQuery', () => {
+  const queryGenerator = sequelize.queryGenerator;
+
+  it('generates a query that adds an index', () => {
+    expectsql(() => queryGenerator.addIndexQuery('table', { fields: ['column1', 'column2'] }), {
+      default: 'CREATE INDEX [table_column1_column2] ON [table] ([column1], [column2])',
+      snowflake: notSupportedError,
+      'mariadb mysql': 'ALTER TABLE `table` ADD INDEX (`column1`, `column2`)',
+    });
+  });
+
+  it('generates a query that adds an index with column using functions', () => {
+    expectsql(() => queryGenerator.addIndexQuery('table', { name: 'myindex', fields: [sequelize.fn('UPPER', sequelize.col('test'))] }), {
```

Comment:
Accessing expression builders on the sequelize instance is deprecated (because it's not necessary, they're fully static). Access them on the `sql` namespace instead: https://sequelize.org/docs/v7/querying/raw-queries/

Using `col` is also discouraged: https://sequelize.org/docs/v7/querying/raw-queries/#sqlcol

Reply:
Updated test now